src/main/java/epam/gym/controller/TrainingController.java

```diff
@@ -1,41 +1,66 @@
 package epam.gym.controller;
 
 import epam.gym.dto.training.TrainingDto;
+import epam.gym.entity.ActionType;
+import epam.gym.entity.Training;
 import epam.gym.service.TrainingService;
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
+import org.springframework.jms.core.JmsTemplate;
 import org.springframework.web.bind.annotation.DeleteMapping;
 import org.springframework.web.bind.annotation.PathVariable;
 import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestBody;
 import org.springframework.web.bind.annotation.RequestMapping;
 import org.springframework.web.bind.annotation.RestController;
 
+import java.util.Optional;
+
+
 @RestController
 @RequestMapping("/trainings")
 public class TrainingController {
 
     private TrainingService trainingService;
 
+    @Value("${activemq.destination}")
+    private String destination;
+
+    private JmsTemplate jmsTemplate;
+
+
     TrainingController(TrainingService trainingService){
         this.trainingService = trainingService;
 
     }
 
     @PostMapping("/add")
     public ResponseEntity<String> addTraining(@RequestBody TrainingDto trainingDto) {
-        trainingService.create(trainingDto);
+        Training training = trainingService.create(trainingDto);
+        trainingService.convertAndSendToConsumer(training, destination, ActionType.ADD);
```

Comment:
I believe it is better to move convertAndSendToConsumer(training, destination, ActionType.ADD); to service layer and remove from controller level